lightning-invoice/src/lib.rs

```diff
@@ -796,17 +796,18 @@ impl SignedRawInvoice {
 ///
 /// The following example would extract the first B.
 /// ```
-/// use Enum::*;
+/// use lightning_invoice::{find_extract, find_all_extract};
 ///
 /// enum Enum {
 /// 	A(u8),
 /// 	B(u16)
 /// }
 ///
-/// let elements = vec![A(1), A(2), B(3), A(4)];
+/// let elements = vec![Enum::A(1), Enum::A(2), Enum::B(3), Enum::A(4)];
 ///
-/// assert_eq!(find_extract!(elements.iter(), Enum::B(ref x), x), Some(3u16));
+/// assert_eq!(find_extract!(elements.iter(), Enum::B(x), x), Some(3u16));
 /// ```
+#[macro_export]
```

Comment:
Not sure if we want to export these though... Is there a way to do it conditionally for tests?

Reply:
Tried `#[cfg_attr(test, macro_export)]`, but that's not working. It'd be a shame to make the code not be evaluated, as it's verifying the accuracy of the comment though.

Reply:
It's even wilder. When I explicitly say `#[cfg_attr(not(test), macro_export)]` as a sanity check, it works!

Reply:
How about `doctest`?

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html?highlight=doctest#include-items-only-when-collecting-doctests

Reply:
Ah, you've found the attribute Google wouldn't show me. Yet alas, also no.

Reply:
So question. If the docs show how to use the macro, wouldn't that require the macro be public? Otherwise the docs are simply inaccurate.

Reply:
I think it was more to test that the macros works as expected and to show internal developers how to use it. It's possible to generate docs for private items, too.

Suppose it could have be a normal test. Maybe already covered by other tests, though. I'd say just remove the ticks as Matt suggested.

Reply:
will do

Reply:
Ok, I gave up on the macros and reverted everything except the semicolons.